---
title: 'The Brazillian Amazon & COVID-19'
author: "Adam Zimmerman"
date: "December 1, 2020"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "/data/sars_vcf_analysis/02_genome_reference/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable_PRJNA662684.txt"
  prevalence_path: "data/raw_data/Seroreversion_adjusted_prevalence.csv"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview

This is a report on SARS-CoV-2 genetic data, as well as some COVID-19 prevalence data for Manaus and Sao Paulo. [@Buss2020.09.16.20194787].

# Methods

See the set of tutorials on the [vcfR package website](https://knausb.github.io/vcfR_documentation/index.html).

You may also want to use any of a range of different COVID data packages and data sources:

* https://kjhealy.github.io/covdata/
* https://github.com/como-ph/oxcovid19
* https://ropensci.org/blog/2020/10/20/searching-medrxivr-and-biorxiv-preprint-data/
* https://covidtracking.com/data/api
  * `readr::read_csv("https://api.covidtracking.com/v1/states/daily.csv")`
* https://rt.live/
  * `readr::read_csv("https://d14wlfuexuxgcm.cloudfront.net/covid/rt.csv")`

## Subsections are ok too

# Results and Discussion

```{r load-packages-and-functions}
library("vcfR")
library("ggplot2")
library("dplyr")
library("ggthemes")
library("medrxivr")
library("oxcovid19")
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

```{r load-oxcovid19-data}
# OxCOVID19 setup
# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)

## Step 1: Create a connection to OxCOVID19 PostgreSQL server
con <- connect_oxcovid19()

## Step 2: Access epidemiology table from OxCOVID19 PostgreSQL server
epi_tab <- get_table(con = con, tbl_name = "epidemiology")

## Step 3: Query the epidemiology table to show data for Brazil
bra_epi_tab <- dplyr::filter(.data = epi_tab, countrycode == "BRA", adm_area_1 == "Amazonas")
```

# Figures

```{r sample-plot}
# create a plot of unique SNP locations within each gene across all samples
vcf_with_metadata %>%
  filter(!is.na(gene)) %>% # get rid of SNPs not in gene regions
  group_by(gene, pos) %>%
  tally() %>% # this gives a column n for the number of gene by position
  group_by(gene) %>%
  tally() %>% # this collapses that down to the number of unique SNP locations
  ggplot(aes(x = gene,
             y = n)) +
    geom_col() +
    labs(title = "Count of distinct SNPs in Named SARS-CoV-2 Genes",
         x = "Gene Name") +
  theme_few() # get rid of the grey background
```

**Figure 1**: N and S genes have more unique SNPs in the set of samples analyzed.

```{r confirmed-plot}
# produce a line plot of confirmed by state, Amazonas or Sao Paulo, over time
ox-plot <- bra_epi_tab %>%
  group_by(date) %>%
  ggplot(aes(x = date,
             y = confirmed)) +
  geom_line() +
  labs(title = "Confirmed COVID-19 cases over time",
       x = "Date",
       y = "Confirmed Cases")

# TODO: Adam could use help in how to setup two cities to compare in one or two plots 
ggsave(plot = confirmed-plot,
       filename = "output/figures/confirmed.png ")
```

**Figure 2: Manaus saw more severe confirmed cases of COVID-19 than Sao Paulo.

```{r prevalence-plot}
# produce a line plot of seroreversion adjusted prevalence of SARS-CoV-2
# antibodies across Manaus and Sao Paulo
prevalence-plot <- prevalence_path %>%
  group_by(Month) %>%
  ggplot(aes(x = Month,
             y = prevalence)) +
  geom_line() +
  labs(title = "Seroreversion adjusted prevalence of SARS-CoV-2 antibodies
                     across Manaus and Sao Paulo",
       x = "Month",
       y = "Antibody prevalence")

ggsave(plot = prevalence-plot,
       filename = "output/figures/prevalence_plot.png")

prevalence-plot
```

**Figure 3: Manaus saw more antibody prevalence of SARS-CoV-2 than Sao Paulo.

# Tables

```{r example-table}
# An example table to show the length of each gene using its start and end
gene_table %>%
  mutate(length = end - start) %>%
  select(gene_name, start, end, length) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Start",
                             "End",
                             "Length"))
```

**Table 1**: Gene names, locations, and lengths in the SARS-CoV-2 genome. Higher SNP counts in the S and N genes may be related to the larger size of these genes.

# Sources Cited
