---
title: 'Accelerated case rates led to herd immunity but also elderly and male mortality in Amazonas, Brazil during the COVID-19 pandemic'
author: "Adam Zimmerman"
date: "December 1, 2020"
output: pdf_document
csl: bioinformatics.csl
bibliography: references.bib
params:
  gff_file_path: "/data/sars_vcf_analysis/02_genome_reference/sars_refgenome_annotation.gff"
  vcf_dir_path: "data/11_vcf_output_for_R"
  sra_runtable_path: "data/00_sra_runtable/SraRunTable_PRJNA662684.txt"
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = TRUE)
```

# Background and Overview

What happened in the Amazon and São Paulo during COVID-19? 

This exploratory analysis of tabular and genetic data shows that the city of Manaus, and its state, Amazonas, were hit harder per capita than the city and state of São Paulo through the second (Q2) and third quarters (Q3) of 2020 \textcolor{red}{TODO: additional citations}. Deaths were especially high for elderly patients, as well as male patients in Manaus. 

From Q2 into Q3, the intense period of infection correlated with the emergence of an antibody prevalence among Manaus blood donors that was approximately threefold greater than antibody prevalence among São Paulo blood donors [@Buss2020.09.16.20194787]. This is an indication of good news and herd immunity for Manaus, heading into Q4.

Additionally, a variant call analysis of 33 SARS-CoV-2 samples showed that the greatest numbers of high quality, distinct SNPs in the named genes originated in the Amazonian states of Maranhao and Para.

# Methods

## Tabular data analysis

\textcolor{red}{TODO: tabular parsing walkthrough: trace piping especially.}

In the final steps of data processing, I temporarily created a column in `Mortality_data.csv` that compared the ratios of mortality figures. I selected the age range 55-60 to show the maximum ratio seen between male and female excess mortality in Manaus this year.

## Genetic data analysis

Illumina NGS runs were collected as 33 .fasta files from the [NCBI SRA website](https://www.ncbi.nlm.nih.gov/sra/). Under the project number PRJNA662684, the runs came grouped by region, being specific to the Brazillian Amazon. 

\textcolor{red}{TODO:Data were downloaded from the NCBI SRA archive from project numbers _ on date. Then processed with trimmomatic and run through a bwa variant calling pipeline (CITATION from bwa paper). … That data was then brought into RStudio to analyze in combination. Include steps of modifying datatables, if applicable.)}

\textcolor{red}{TODO: sequencing and bash walkthrough: trace scripts as sets by format; citations} 

Lastly, single nucleotide polymorphisms (SNPs) were tallied and grouped by region, gene, and quality (Table 1).

# Results and Discussion

## Tabular data analysis

This analysis indicates that living in Manaus, being elderly, and being male were risk factors for Brazillians facing COVID-19 in Q2 and Q3 of 2020. It surprised me to see that male patients in Manaus were at such markedly higher risk of death than female patients, to the order of \textcolor{red}{TODO: insert ratio}.

\textcolor{red}{TODO: tabular parsing; refer to figures; 66% of Manaus blood donors versus 22% of São Paulo blood donors. Captions, good axes, citations
Fix fill for col}.

Excess mortality in 50-55 year old male patients reached a maximum of 126 deaths/1000, 1.30 times their expected mortality of 99 deaths/1000, as well as 2.49 times the excess mortality of female patients of the same age at 54.8 deaths/1000 (Figure 6).

## Genetic data analysis

The variant call analysis resulted in a set of __ high quality, distinct SNPs. The notable majority of these SNPs in named genes came from Maranhao and Para, as opposed to the other nearby Amazonian states (Table 1). 

\textcolor{red}{TODO: fastqc screenshot: evidence that it’s reasonable
Quality, length, WATCH a few weeks ago on fastqc examples with one goofy one
, citations}.

 ```{r load-packages-and-functions}
# These functions are also listed in DESCRIPTION for .travis.yml
library("vcfR")
library("ggplot2")
library("dplyr")
library("tidyr")
library("readr")
library("ggthemes")
library("medrxivr")
library("oxcovid19")
library("lubridate")

# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)
```

```{r load-vcf-data}
# load in, tidy, and stack all of the VCF files in the target directory
# these are the output of the bash pipeline
stacked_vcfs <- parse_tidy_and_stack_vcfs(
  vcf_dir_path = params$vcf_dir_path)

# load in the gff file with genome annotations for SARS-CoV-2
gff <- read_gff(gff_file_path = params$gff_file_path)

# pull out just the named genes from the annotation file
gene_table <- extract_genes_from_gff(annotation_object = gff)

# combine the stacked vcf table with the gene names and the metadata from
# the sra runtable file
vcf_with_metadata <- add_genes_metadata_to_vcfstack(
  sra_runtable_path = params$sra_runtable_path,
  stacked_vcf = stacked_vcfs,
  cleaned_genes_table = gene_table)
```

```{r load-confirmed-data}
# OxCOVID19 confirmed cases setup
# this loads all function files in the code/functions directory
function_paths <- list.files("code/functions", full.names = TRUE)
purrr::walk(function_paths, source)

## Step 1: Create a connection to OxCOVID19 PostgreSQL server
con <- connect_oxcovid19()

## Step 2: Access epidemiology table from OxCOVID19 PostgreSQL server
epi_tab <- get_table(con = con, tbl_name = "epidemiology")

## Step 3: Query the epidemiology table to show data for Brazil
bra_epi_tab <- dplyr::filter(.data = epi_tab, countrycode == "BRA", adm_area_1 
                             %in% c("Amazonas", "São Paulo"))
# readr::write_csv(bra_epi_tab, "./output/confirmed.csv")
```

# Figures

```{r confirmed-plot}
# produce a colored line plot of confirmed cases by state
confirmed_plot <- bra_epi_tab %>%
  group_by(date) %>%
  ggplot(aes(x = date,
             y = confirmed)) +
  geom_line(aes(colour=adm_area_1)) +
  labs(title = "Confirmed COVID-19 cases over time",
       x = "Date",
       y = "Confirmed Cases") +
  theme_clean()

confirmed_plot

ggsave(plot = confirmed_plot,
       filename = "output/figures/confirmed_plot.png")
```

**Figure 1**: The state of São Paulo saw more total confirmed cases than the state of Amazonas.
\newpage

```{r population-plot}
# produce a pie chart highlighting São Paulo and Amazonas vs Total
population_path <- readr::read_csv("data/raw_data/Brazil_populations.csv")
population_data <- population_path %>%
  group_by(population) %>%
  dplyr::filter(RegionName %in% c("Amazonas", "São Paulo", "Total"))

# Remove Amazonas and São Paulo populations from Total
population_data[3,3] <- (population_data[3,3] - population_data[2,3] - population_data[1,3])

# Basic piechart
population_plot <- ggplot(population_data, aes(x="", y=population, fill=RegionName)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=0) +
  labs(title = "Population of Brazil") +
  theme_void() # remove background, grid, numeric labels

population_plot

ggsave(plot = population_plot,
       filename = "output/figures/population_plot.png")
```

**Figure 2**: The state of São Paulo is much larger than Amazonas by population at 46 million versus 4 million.
\newpage

```{r per-capita-plot}
# produce a dual line plot of confirmed cases by state, Amazonas or São Paulo
per_capita_amazonas <- bra_epi_tab %>%
  dplyr::filter(adm_area_1 == "Amazonas") %>%
  group_by(date) %>%
  dplyr::mutate(per_capita_a = confirmed/4207714) %>%
  dplyr::select(date, per_capita_a)

per_capita_sp <- bra_epi_tab %>%
  dplyr::filter(adm_area_1 == "São Paulo") %>%
  group_by(date) %>%
  dplyr::mutate(per_capita_s = confirmed/46289333) %>%
  dplyr::select(date, per_capita_s)

# per_capita_amazonas
# per_capita_sp
per_capita_plot <- ggplot() +
  geom_line(data=per_capita_amazonas, aes(x = date, y = per_capita_a, color = "Amazonas")) +
  geom_line(data=per_capita_sp, aes(x = date, y = per_capita_s, color = "São Paulo")) +
  labs(title = "Confirmed COVID-19 cases per capita over time",
       x = "Date",
       y = "Confirmed Cases Per Capita",
       color = "State") +
  theme_clean()

per_capita_plot

ggsave(per_capita_plot,
       filename = "output/figures/per_capita_plot.png")
```

**Figure 3**: Amazonas saw greater confirmed cases per capita than São Paulo.
\newpage

```{r prevalence-plot}
# produce a dual-line plot of seroreversion adjusted prevalence of SARS-CoV-2
# antibodies across Manaus and São Paulo
prevalence_data <- readr::read_csv("data/raw_data/Adjusted_prevalence.csv") %>%
  # lubridate::month(date) %>%
  tidyr::pivot_longer(!date, names_to="region")
prevalence_plot <- prevalence_data %>%
  ggplot() +
  geom_line(aes(x = date, y = value, color = region)) +
  labs(title = "Prevalence of SARS-CoV-2 antibodies (seroreversion adjusted)",
       x = "Month",
       y = "Antibody prevalence (% of population)") +
  scale_color_manual("City",
                     labels = c("Manaus", "São Paulo"),
                     values = c("cyan", "red")) +

  theme_clean()

ggsave(plot = prevalence_plot,
       filename = "output/figures/prevalence_plot.png")

prevalence_plot
```


**Figure 4**: Manaus saw more widespread prevalence of SARS-CoV-2 antibodies vs. São Paulo, up to 66% vs. 22% of the population respectively.
\newpage

```{r mortality-plot-f}
# create a bar chart of female mortality in Manaus, March - June
mortality_data_f <- readr::read_csv("data/raw_data/Excess_mortality.csv") %>%
  select(-observed_mort_f, -excess_mort_total,
         -excess_mort_percent, -observed_mort_m, -excess_mort_m, -expected_mort_m) %>%
  tidyr::pivot_longer(!age, names_to="mortality")


mortality_plot_f <- ggplot(data = mortality_data_f, aes(x = age, y = value, color = mortality)) +
  geom_col(aes(color = mortality, fill = mortality)) +
  labs(title = "Female mortality in Manaus, March 2020 to June 2020",
       x = "Age",
       y = "Deaths per 1000") +
  # scale_x_discrete(position = "")
  scale_color_manual("Patient mortality classification",
                     labels = c("Excess mortality female",
                                "Expected mortality female"),
                     values = c("coral", "darkseagreen4")) +

  theme_clean() +
  
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

mortality_plot_f

ggsave(mortality_plot_f,
       filename = "output/figures/mortality_plot_f.png")

```

**Figure 5**: Excess mortality, unexpected deaths of any cause, reached a maximum of 2.49 times the expected mortality in female patients.
\newpage

```{r mortality-plot-m}
# create a bar chart of male mortality in Manaus, March - June
mortality_data_m <- readr::read_csv("data/raw_data/Excess_mortality.csv") %>%
  select(-observed_mort_f, -excess_mort_total,
         -excess_mort_percent, -observed_mort_m, -excess_mort_f, -expected_mort_f) %>%
  tidyr::pivot_longer(!age, names_to="mortality")


mortality_plot_m <- ggplot(data = mortality_data_m, aes(x = age, y = value, color = mortality)) +
  geom_col(aes(color = mortality, fill = mortality)) +
  labs(title = "Male mortality in Manaus, March 2020 to June 2020",
       x = "Age",
       y = "Deaths per 1000") +
  # scale_x_discrete(position = "")
  scale_color_manual("Patient mortality classification",
                     labels = c("Excess mortality male",
                                "Expected mortality male"),
                     values = c("coral", "darkseagreen4")) +

  theme_clean() +
  
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

mortality_plot_m

ggsave(mortality_plot_m,
       filename = "output/figures/mortality_plot_m.png")

```

\textcolor{red}{TODO: unique caption}
\newpage

# Tables

```{r genetic-table}
# Create a table showing what genes had the greatest SNPs, where in Brazil
vcf_with_metadata %>%
  group_by(geo_loc_name, qual, gene) %>%
  tally() %>%
  select(gene, geo_loc_name, qual) %>%
  knitr::kable(col.names = c("Gene Name",
                             "Brazillian State",
                             "Quality"))
```

**Table 1**: The greatest numbers of high quality, distinct SNPs in the named genes came from Maranhao and Para, Brazil.

# Sources Cited
